### PR TITLE
Feat/markdown, html, tsv and psv

### LINKED ISSUE #190: Framework : Add HTML, Markdown, PSV, TSV data format support for HTTP consumption
### Component

Core Engine

### Problem & Motivation

As a capability designer, I can't currently consume APIs that return Markdown content in a structured way. The `outputRawFormat` field on `ConsumedHttpOperation` supports `json`, `xml`, `avro`, `protobuf`, `csv`, and `yaml` — but not `markdown`.

Markdown is increasingly used as a machine-readable content format beyond documentation:

- GitHub API (`/repos/.../readme`) returns Markdown bodies with front matter and tables
- CMS platforms (Strapi, Contentful) serve Markdown content fields
- Static-site repos expose `.md` files with YAML front matter
- LLM/AI tool outputs are frequently Markdown-formatted
- Wiki exports (Notion, Confluence) produce Markdown with heading structure and tables

In all cases, structured data (tables, front matter metadata, sectioned content) is trapped inside Markdown syntax. Today, extracting it requires custom code outside the framework — breaking the pure-YAML capability authoring experience.

### Proposed Solution

Add `"markdown"` to the `outputRawFormat` enum, backed by a lightweight Markdown parser (commonmark-java — ~150 KB, CommonMark-compliant + GFM tables extension).

The converter parses Markdown into a predictable JSON tree with three top-level keys:

- `$.frontMatter` — YAML front matter parsed to a JSON object (empty object `{}` if absent)
- `$.tables[N]` — each pipe-delimited GFM table as an array of objects (header row → keys)
- `$.sections[N]` — heading-based sections with `heading`, `level`, and `content` (plain text)

The existing JSONPath mediation layer handles extraction and mapping unchanged — same developer experience as `csv`, `xml`, or `yaml`.

**In scope:** tables, YAML front matter, heading-based sections, plain text content.
**Not in scope (v1):** code block extraction, link extraction, nested list objects, column type inference.

### Sister proposal
This proposal is paired with a HTML output format proposal, which uses the same `$.tables[N]` JSON shape. JSONPath expressions for table extraction are intentionally interchangeable between the two formats — a capability author can switch between an HTML and a Markdown source by changing only `outputRawFormat`.

Recommend bundling both in a single PR to enforce the shared `tables` convention.

### Implementation changes
| Component | Change |
|---|---|
| `pom.xml` | Add `commonmark:0.24.x` + `commonmark-ext-gfm-tables:0.24.x` |
| `Converter.java` | Add `convertMarkdownToJson(Reader)` + `"MARKDOWN"` branch |
| `naftiko-schema.json` | Add `"markdown"` to `outputRawFormat` enum |
| `ConverterTest.java` | Unit tests: table parsing, front matter, sections, edge cases |
| `MarkdownIntegrationTest.java` | End-to-end test with mock Markdown endpoint |
### Example

```yaml
consumes:
  - type: http
    namespace: github
    baseUri: https://raw.githubusercontent.com
    resources:
      - name: readme
        path: /naftiko/framework/main/README.md
        operations:
          - name: get-readme
            method: GET
            outputRawFormat: markdown
            outputParameters:
              - name: title
                type: string
                value: $.frontMatter.title
              - name: version
                type: string
                value: $.frontMatter.version
              - name: featureTable
                type: array
                value: $.tables[0]
              - name: breakingChanges
                type: string
                value: "$.sections[?(@.heading=='Breaking Changes')].content"
```

### Agent Context (optional)

agent_name: Notion AI
tool: Notion AI — chat
llm_model: claude-sonnet-4-6
confidence: high
source_event: user request via Notion chat — based on ADR "Markdown Output Format Support Proposal" (Draft, v1.0 Alpha 1)
note: Feature request revised after review of internal design doc. HTML sister proposal should be filed separately and both PRs bundled.

### PR DESCRIPTION
**Pull request by jlouvel**
## Related Issue

Closes #190

---

## What does this PR do?

This PR adds native support for additional HTTP consumed response formats and updates related docs/examples/tests.

Main changes:
- Adds new output raw formats:
  - html
  - markdown
  - tsv (tab-separated)
  - psv (pipe-separated)
- Refactors delimited parsing in the converter:
  - Introduces shared delimiter-based conversion
  - Keeps csv behavior backward-compatible
- Extends schema/docs:
  - Updates outputRawFormat enum/documentation in schema docs/wiki/README
  - Adds mention of TSV/PSV and HTML/Markdown in relevant markdown/wiki pages
- Adds examples and fixtures:
  - New capability examples for HTML and Markdown
  - New sample fixtures for HTML/Markdown/TSV/PSV
- Adds tests:
  - Converter unit tests for HTML/Markdown and delimited formats
  - Integration tests for HTML and Markdown consumption

Also includes roadmap/documentation alignment updates for the new format support.

---

## Tests

Executed:
- mvn -ntp test "-Dtest=ConverterTest,CsvIntegrationTest"

Result:
- BUILD SUCCESS
- ConverterTest and CsvIntegrationTest pass with the new format coverage.

Note:
- Full suite has pre-existing unrelated tutorial failures in Step*ShipyardMcpClientIntegrationTest classes (not introduced by this PR).

---

## Checklist

- [ ] CI is green (build, tests, schema validation, security scans)
- [x] Rebased on latest `main`
- [x] Small and focused — one concern per PR
- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/)

---

## Agent Context (optional)

```yaml
agent_name: GitHub Copilot
llm: GPT-5.3-Codex
tool: VS Code Copilot coding agent
confidence: high
source_event: user_request
discovery_method: user_report
review_focus: Converter.java, naftiko-schema.json, ConverterTest.java, docs/wiki updates
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -139,6 +139,21 @@
...
+        <dependency>
+            <groupId>org.commonmark</groupId>
+            <artifactId>commonmark</artifactId>
+            <version>0.24.0</version>
+        </dependency>
+        <dependency>
+            <groupId>org.commonmark</groupId>
+            <artifactId>commonmark-ext-gfm-tables</artifactId>
+            <version>0.24.0</version>
+        </dependency>
```
Comment (eskenazit) [inline pom.xml]: There does not seem to be any import of commonmark in the code

File: src/main/java/io/naftiko/engine/Converter.java [hunk 2]
```diff
@@ -340,4 +424,276 @@ public static String fixJsonPathWithSpaces(String mapping) {
...
+            return false;
+        }
+
+        for (String cell : parseMarkdownCells(line)) {
+            String trimmed = cell.trim();
+            if (trimmed.isEmpty()) {
+                continue;
+            }
+            if (!trimmed.matches("^:?-{3,}:?$") && !trimmed.matches("^-{3,}$")) {
+                return false;
```
[the remark this answers is not in the document]
Comment (jlouvel) [reply src/main/java/io/naftiko/engine/Converter.java]: Fixed

File: src/main/java/io/naftiko/engine/Converter.java [hunk 3]
```diff
@@ -137,6 +157,70 @@ public static JsonNode convertCsvToJson(Reader csvReader) throws IOException {
...
+    }
+
+    /**
+     * Convert Markdown input to JsonNode and extract front matter, tables, and sections.
+     *
+     * @param markdownReader Reader containing Markdown data
+     * @return JsonNode with frontMatter, tables, and sections
+     * @throws IOException if parsing fails
+     */
+    public static JsonNode convertMarkdownToJson(Reader markdownReader) throws IOException {
```
[the remark this answers is not in the document]
Comment (jlouvel) [reply src/main/java/io/naftiko/engine/Converter.java]: Fixed
